### ISSUE TITLE
External linter shows misleading errors

### ISSUE THREAD

**Issue by vlad20012** [intellij-rust/intellij-rust#10429]
## Environment

* **IntelliJ Rust plugin version:** 0.4.194.SNAPSHOT-231-dev
* **Rust toolchain version:** 1.69.0 (84c898d65 2023-04-16) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2023.1 Ultimate Edition (IU-231.8109.175)
* **Operating system:** Linux 5.19.0-38-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal

## Problem description

![image](https://user-images.githubusercontent.com/3221931/236208336-952250d6-9b6f-403e-8c67-713f9d8b6a8b.png)

## Steps to reproduce
1. Enable `cargo check` external linter
2.
```rust
struct S;
fn main() {
    let s = S;
    let _a = s;
    let _b = s;
}
```

**Comment by vlad20012** [comment]
Another example:
```rust
struct S;

fn main() {
    a;
}
```

![image](https://user-images.githubusercontent.com/3221931/236208978-aef030f6-0e61-4b6a-abff-693cb19aea7b.png)

![image](https://user-images.githubusercontent.com/3221931/236209036-0967ceb6-1628-4d9a-9d92-ede449bceaa8.png)
